ThunderCloud/PopupView.swift

```diff
@@ -0,0 +1,340 @@
+            
+        ] + buttonConstraints() + spaceViewConstraints())
+    }
+    
+    /// `NSLayoutConstraint`s for `UIButton`s
+    private func buttonConstraints() -> [NSLayoutConstraint] {
+        return [confirmButton, cancelButton].flatMap {[
+            $0.heightAnchor.constraint(equalToConstant: Constants.buttonHeight),
+            $0.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+        ]}
+    }
+    
+    /// `NSLayoutConstraint`s for `spaceView`s
+    private func spaceViewConstraints() -> [NSLayoutConstraint] {
+        return Space.allCases.flatMap {[
+            spaceViews[$0.rawValue].heightAnchor.constraint(equalToConstant: $0.height),
+            spaceViews[$0.rawValue].widthAnchor.constraint(equalTo: stackView.widthAnchor)
+        ]}
+    }
+       
+    // MARK: - UI
+    
+    /// `PopupConfig` `didSet`
+    private func didUpdatePopupConfig() {
+        imageView.image = config.image
+        titleLabel.text = config.title
+        subtitleLabel.text = config.subtitle
+        detailLabel.text = config.detail
+        confirmButton.setTitle(config.confirmText, for: .normal)
+        cancelButton.setTitle(config.cancelText, for: .normal)
+    }
+    
+    // MARK: - UIControlEvent
+    
+    /// `confirmButton` `.touchUpInside`
+    @objc func confirmButtonTouchUpInside(_ sender: UIButton) {
+        delegate?.popupView(self, confirmButtonTouchUpInside: sender)
+    }
+    
+    /// `cancelButton` `.touchUpInside`
+    @objc func cancelButtonTouchUpInside(_ sender: UIButton) {
+        delegate?.popupView(self, cancelButtonTouchUpInside: sender)
+    }
+    
+    /// Catch any tap gestures
+    @objc func viewTapped(_ sender: UITapGestureRecognizer) {
+        // do nothing
+    }
+    
+    // MARK: - General
+    
+    private static func createLabel(font: UIFont) -> UILabel {
+        let label = UILabel()
+        label.textColor = .black
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.font = font
+        return label
+    }
+    
+    private func createButton(selector: Selector, titleColor: UIColor, backgroundColor: UIColor) -> UIButton {
+        let button = UIButton()
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 18, weight: .semibold)
+        button.setTitleColor(titleColor, for: .normal)
+        button.backgroundColor = backgroundColor
+        button.addTarget(self, action: selector, for: .touchUpInside)
+        button.clipsToBounds = true
+        button.layer.cornerRadius = Constants.buttonCornerRadius
+        return button
+    }
+}
+
+// MARK: - Extensions
+
+extension UIImage {
+    
+    /// Tick image defined in Storm.xcassets
+    static let tick = UIImage(named: "tick", in: Bundle(for: PopupView.self), compatibleWith: nil)
+}
+
+extension UIColor {
+    
+    static let red = UIColor(red: CGFloat(237)/255, green: CGFloat(27)/255, blue: CGFloat(46)/255, alpha: 1)
```

Comment:
Let's use `Theme` instead of these plz :D Should use `mainColor` in replacement of `red` and then possibly `lightGray` (If that exists) and `darkGray` (If that exists)